lighthouse-core/gather/gatherers/seo/robots-txt.js

```diff
@@ -8,13 +8,13 @@
 const FRGatherer = require('../../../fraggle-rock/gather/base-gatherer.js');
 const {getBrowserVersion} = require('../../driver/environment.js');
 
-/* global fetch, location */
+/* global fetch, location, window */
 
 /** @return {Promise<LH.Artifacts['RobotsTxt']>} */
 /* c8 ignore start */
 async function getRobotsTxtContent() {
   try {
-    const response = await fetch(new URL('/robots.txt', location.href).href);
+    const response = await fetch(new window.__nativeURL('/robots.txt', location.href).href);
```

Comment:
took awhile to figure out why the smoke test wasn't passing, it's because seo tester html overrides `URL`! so, luckily we get a driveby fix here for naughty pages overriding globals.

Reply:
hm, `_cachedNativesPreamble` should be solving this, is something broken there? that's more concerning than the smoke itself :/

https://github.com/GoogleChrome/lighthouse/blob/ce870607f8bbcabdaf3292b30f759e69f36cb9c3/lighthouse-core/gather/driver/execution-context.js#L226-L235

Reply:
ok so this seems fine and i can't get it to fail again, weird